io/eolearn/io/sentinelhub_process.py

```diff
@@ -66,7 +75,13 @@ def __init__(self, data_collection, size=None, resolution=None, cache_folder=Non
         self.data_collection = DataCollection(data_collection)
         self.cache_folder = cache_folder
 
-    def execute(self, eopatch=None, bbox=None, time_interval=None):
+    def execute(
+        self,
+        eopatch=None,
+        bbox=None,
+        time_interval=None,
+        session_obtainer: Optional[Callable[[], SentinelHubSession]] = None,
```

Comment:
As discussed, I have 2 comments here:

- Conceptually it might be better that `session_obtainer` is an `__init__` parameter than an `execute` parameter. This is mainly because I don't see a use case where you would have to provide a different session obtainer for different spatial and temporal parameters. It seems that a task in a workflow will always require the same object here.

- The word "obtainer" sounds a bit weird to me. Maybe name `session_loader` would be more natural?

Reply:
For the first point, the only reason I see why `exec_args` are better is that you can initialize the session sharing object after the workflow is built. In case of ray, one must first create the actor before one can create a `session_obtainer`

Loader has a I/O feel to me, but honestly I don't mind. Will change

Reply:
Looks good.